tensorflow/contrib/cmake/python_modules_test.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+
+from tensorflow.python.platform import test
+
+
+entries_file = "tensorflow/contrib/cmake/python_modules.txt"
```

Comment:
you can use the resource_loader library to access this file.

Reply:
You mean `self.entries = tf.resource_loader.load_resource(entries_file).split("\n")` ?

Reply:
Trying to remove dependencies to other packages so as not to import from `runfiles` which fails if something references generated files.

Reply:
Since I need to compute absolute paths elsewhere in the script and `resource_loader` has its quirks, I have adapted the code and placed it directly in the tests.

Reply:
I am afraid you will need to depend on runfiles, and nothing else for a test that we want to run in an automated way. Because the way we are using bazel is, your test is only guaranteed to have full access to runfiles, and it "won't" have access to the original source tree.
And also, I would recommend going through resource loader. This is because the runfiles location/structure may change. We will maintain resource_loader ourselves to ensure it will keep working the same way it did work before.
If you do not use resource_loader, it is just a 2nd place to maintain this logic. It will be brittle, as runfiles is managed by bazel, and bazel is still not 1.0.

Reply:
The only thing that would be needed is to extract the path resolution into a separate function. This should make everyone happy.

Reply:
Also, I am aware that only `runfiles` are guaranteed to exist. That is exactly why I introduced the hack I mentioned.